### ISSUE TITLE
Entity ranking

### ISSUE THREAD

**Issue by tabergma** [RasaHQ/rasa#3854]
**Description of Problem**:
related to https://forum.rasa.com/t/entity-not-recogised-properly/11700/4

You trained your NLU pipeline and you wonder why a specific entity is not picked up by the NER. You can test your model, however, there is no file showing you the errors your NER made. Also you don't see any kind of entity ranking. So, overall it is hard to figure out, why the NER misses a specific entity. 

**Overview of the Solution**:
- When testing a model, also write down errors/successes of the NER.
- If possible, provide some kind of ranking for entities. What entities were not applied due to low confidence?

**Definition of Done**:
- [ ] Tests are added
- [ ] Feature is documented
- [ ] Entity ranking was discussed and either implemented or ruled out.
- [ ] Errors/Successes of NER are written to a file during testing.

**Comment by wochinge** [comment]
@tabergma Will you tackle this as part of https://github.com/RasaHQ/research/issues/51 cause you need if for testing anyway?

**Comment by tabergma** [comment]
@wochinge Yes. We will discuss it and decide if it makes sense or not. If we decide to implement it, I'll do it along the other improvements of the NER.

**Comment by Ghostvv** [comment]
Isn't NER prints the errors in the file?

**Comment by tabergma** [comment]
Yes, added that at some point. Still no ranking, but nor sure if we want to add it.

**Comment by Ghostvv** [comment]
adding scores in in the new architecture is fairly straightforward. I say let's add scores while we're on it

**Comment by tabergma** [comment]
@Ghostvv I don't think that we actually need an entity ranking, for example: 
`I am flying from Berlin to London` - `entities: "Berlin" - departure; ranking: [departure, destination, none]`
I am not sure if that gives us any benefit. What do you think?

**Comment by Ghostvv** [comment]
similar to intent ranking, I guess. However with crf it is a bit hard to do, since picked entities maximize the sequence

**Comment by tabergma** [comment]
The question is: Do we want to implement that at one point or should we close the issue?

**Comment by Ghostvv** [comment]
I think in principle it makes sense to have a ranking, but due to complexity of implementation, I suggest to deprio this issue

**Comment by samsucik** [comment]
@Ghostvv  @alopez if we deprio this issue, is there some way for us to still keep in mind that we want to introduce entity ranking once our algorithm allows for it? (Given that currently our CRF doesn't really allow this.) We all certainly agree that entity ranking as such is a good thing and, if possible, should be introduced in the future.

**Comment by Ghostvv** [comment]
I don't think that needs to be an issue, but it'd be good to keep it somewhere

**Comment by koernerfelicia** [comment]
Related: https://github.com/RasaHQ/rasa/issues/7299

**Comment by alexberman** [comment]
Here's some input regarding the prioritization of this issue. Entity ranking (with confidence estimates) is needed for identifying genuine ambiguities relating to entities. In a domain that we are currently working with, an entity such as "Paris" can either refer to a city or a person name. Ideally, disambiguition in such cases should be done by the dialog manager, e.g. by using dialog context or by asking a follow-up question to the user ("Do you mean the city Paris or the name of a person?") Only getting the top-ranked entity from Rasa excludes these possibilities.

**Comment by Ghostvv** [comment]
@alexberman `crf` (the loss that we use to tag entities) doesn't allow to calculate entity ranking, because second best sequence is not a well defined concept
